### PR TITLE
Added email verification

### LINKED ISSUE #30: Email Verification Script - Dashboard
### Tasks

- [ ] Implement an email verification process using AWS SES.
- [ ] Generate and send a verification email with a unique token.
- [ ] Create a verification endpoint to confirm the email address using the token.

### User Story

As a user, I want to verify my email address to ensure I receive notifications for price changes and sales.

### PR DESCRIPTION
**Pull request by A-Faris**
When the Sign In button is clicked, a verification email is sent for the user to confirm
closes #30

### CODE REVIEW

File: Dashboard/login.py [hunk 1]
```diff
@@ -84,8 +86,13 @@ def login_page(config: _Environ, email_pattern: str) -> None:
...
         elif re.match(email_pattern, signup_email):
-            add_email(conn, signup_email)
-            login(signup_email)
+            client = get_ses_client(config)
+            send_verification_email(client, signup_email)
+            logging.warning("Check your inbox to confirm your email address")
+            st.warning("Check your inbox to confirm your email address")
+            if signup_email in get_ses_emails(client, "verified"):
+                add_email(conn, signup_email)
+                login(signup_email)
```
Comment (BerkayDur) [inline Dashboard/login.py:95]: Nice use of the `utilities` directory.

File: Dashboard/utilities/email_verification.py
Comment (BerkayDur) [inline Dashboard/utilities/email_verification.py:1]: No changes here except formatting.

**Review by BerkayDur — APPROVED** [review]

File: Dashboard/utilities/email_verification.py [hunk 2]
```diff
@@ -19,10 +20,12 @@ def get_ses_client(config: _Environ) -> ses_client:
         region_name = config['AWS_REGION_NAME']
     )
 
+
 def is_ses(boto_ses_client: ses_client) -> bool:
     '''Returns true if ses client else false.'''
     return (isinstance(boto_ses_client, BaseClient)
-            and boto_ses_client._service_model.service_name == 'ses') #pylint: disable=protected-access
+            and boto_ses_client._service_model.service_name == 'ses')  # pylint: disable=protected-access
```
Comment (danfh00) [inline Dashboard/utilities/email_verification.py:27]: Nice use of pylint disabling

**Review by danfh00 — APPROVED** [review]
Looks great!

**Pull request merged**